### PR TITLE
[metadata] Compiler warnings fixes.

### PR DESCRIPTION
**Pull request by nsivov**

### CODE REVIEW

File: mono/metadata/marshal-ilgen.c [hunk 1]
```diff
@@ -4253,28 +4269,6 @@ emit_marshal_custom_ilgen (EmitMarshalContext *m, int argnum, MonoType *t,
 	mklass = mono_class_from_mono_type (mtype);
 	g_assert (mklass != NULL);
 
-handle_exception:
```
Comment (jaykrell) [inline mono/metadata/marshal-ilgen.c]: Isn't this is a significant semantic change? It was previously falling into the label, or goto'ing it. Now the falling is missed.

**Comment by lambdageek** [comment]
Resolved conflicts with `master`.

@nsivov make sure to pull from github if you plan to do any more work on this PR, to pick up the merge commit.

**Comment by lambdageek** [comment]
@monojenkins build  powerpc

File: mono/metadata/marshal-ilgen.c [hunk 2]
```diff
@@ -5857,7 +5857,7 @@ emit_managed_wrapper_ilgen (MonoMethodBuilder *mb, MonoMethodSignature *invoke_s
 		m->vtaddr_var = mono_mb_add_local (mb, int_type);
 
 	ex_local = mono_mb_add_local (mb, m_class_get_byval_arg (mono_defaults.uint32_class));
-	e_local = mono_mb_add_local (mb, m_class_get_byval_arg (mono_defaults.exception_class));
+	mono_mb_add_local (mb, m_class_get_byval_arg (mono_defaults.exception_class));
```
[the remark this answers is not in the document]
Comment (jaykrell) [reply mono/metadata/marshal-ilgen.c]: The function appears to be used.

[the remark this answers is not in the document]
Comment (lambdageek) [reply mono/metadata/marshal-ilgen.c]: Hm. yea: `e_local` and `ex_local` are both leftover from b20d5ce1e5d69512242dd42ae6ff1afc2483c16d when we moved the exception manipulation in this wrapper from the wrapper to the unwinder.  `e_local` and `ex_local` are both unused now.

[the remark this answers is not in the document]
Comment (lambdageek) [reply mono/metadata/marshal-ilgen.c]: I made a separate PR to remove the unused IL vars and the code that initialized `ex_local` - #11379

**Comment by lambdageek** [comment]
@monojenkins build Linux PPC64 little endian

**Comment by lambdageek** [comment]
@monojenkins build failed

**Comment by lambdageek** [comment]
@monojenkins build Linux PPC64 little endian

**Comment by lambdageek** [comment]
@monojenkins rebase

**Comment by nsivov** [comment]
@lambdageek thank you

File: mono/metadata/marshal-ilgen.c (hunk 1 above)
Comment (jaykrell) [reply mono/metadata/marshal-ilgen.c]: I see now. This was maybe indentation reduction, or hot/cold split for efficiency, or human wanted exceptional path out of the way for easy reading, or mis-maintenance.

**Pull request merged**